### PR TITLE
Add Alipay+ wallets and platform type field to the SDK

### PR DESCRIPTION
**Pull request by hpariwat**
* Add 6 currently available Alipay+ wallets: Alipay, AlipayHK, DANA, GCash, Kakao Pay, and TNG eWallet (Touch 'n Go) to the SDK.
* Make it possible to display footnote under each payment method if needed.
* Add new source type field which has "android" as its default value.

### CODE REVIEW

**Review by kan98 — APPROVED** [review]

**Review by pedro-omise — APPROVED** [review]

File: sdk/src/main/java/co/omise/android/ui/OmiseListFragment.kt [hunk 1]
```diff
@@ -106,6 +108,12 @@ class OmiseItemViewHolder(val view: View, val listener: OmiseListItemClickListen
             listItemView.alpha = 0.2F
         }
 
+        if (subtitleText.text != "") {
```
Comment (nuxzero) [inline sdk/src/main/java/co/omise/android/ui/OmiseListFragment.kt:111]:
```suggestion
        if (subtitleText.text.isNotBlank()) {
```

I think proper way to check empty string. https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/is-not-blank.html

File: sdk/src/main/java/co/omise/android/ui/PaymentMethodResources.kt [hunk 2]
```diff
@@ -136,6 +137,54 @@ internal sealed class PaymentMethodResource(
...
+            titleRes = R.string.payment_method_alipay_cn_title,
+            subtitleRes = R.string.payment_method_alipayplus_footnote,
+            indicatorIconRes = R.drawable.ic_redirect,
+            sourceType = SourceType.AlipayCn
+    )
+
+    object AlipayHk : PaymentMethodResource(
+            iconRes = R.drawable.payment_alipay_hk,
+            titleRes = R.string.payment_method_alipay_hk_title,
+        subtitleRes = R.string.payment_method_alipayplus_footnote,
```
Comment (nuxzero) [inline sdk/src/main/java/co/omise/android/ui/PaymentMethodResources.kt:151]: Could you change an indent here?

File: sdk/src/main/res/layout/list_item.xml [hunk 3]
```diff
@@ -16,15 +16,32 @@
...
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="@dimen/medium_margin"
+            android:layout_marginLeft="@dimen/medium_margin"
+            android:layout_weight="1"
+            android:id="@+id/text_with_subtext_view">
+        <TextView
+                android:id="@+id/text_item_title"
+                style="?android:attr/itemTextAppearance"
+                android:layout_width="match_parent"
```
Comment (nuxzero) [inline sdk/src/main/res/layout/list_item.xml:30]:
```suggestion
                android:layout_width="wrap_content"
```

File: sdk/src/main/res/layout/list_item.xml [hunk 4]
```diff
@@ -16,15 +16,32 @@
...
+                android:layout_height="0dp"
+                android:layout_marginStart="@dimen/medium_margin"
+                android:layout_marginLeft="@dimen/medium_margin"
+                android:layout_weight="3"
+                tools:text="Credit Card" tools:ignore="NestedWeights"/>
+        <TextView
+                android:id="@+id/subtext_item_title"
+                style="?android:attr/itemTextAppearance"
+                android:layout_width="match_parent"
```
Comment (nuxzero) [inline sdk/src/main/res/layout/list_item.xml:39]:
```suggestion
                android:layout_width="wrap_content"
```

File: sdk/src/main/res/layout/list_item.xml [hunk 5]
```diff
@@ -16,15 +16,32 @@
...
+                android:layout_height="0dp"
+                android:layout_marginStart="@dimen/medium_margin"
+                android:layout_marginLeft="@dimen/medium_margin"
+                android:layout_weight="3"
+                tools:text="Credit Card" tools:ignore="NestedWeights"/>
+        <TextView
+                android:id="@+id/subtext_item_title"
+                style="?android:attr/itemTextAppearance"
```
Comment (nuxzero) [inline sdk/src/main/res/layout/list_item.xml:38]: This style duplicated with the title text. So, subtitle might looks the same with the title text.

```suggestion
                style="?android:attr/textAppearanceListItemSecondary"
```

However, please test this with different themes to test how's it look like.
 https://developer.android.com/reference/android/R.attr#textAppearanceListItemSecondary

**Review by tookaa2 — APPROVED** [review]

**Review by nikolack — APPROVED** [review]

File: sdk/src/main/res/layout/list_item.xml (hunk 5 above)
Comment (hpariwat) [reply sdk/src/main/res/layout/list_item.xml]: @nuxzero Your suggestion already looks good, thanks.

**Review by nuxzero — APPROVED** [review]

**Pull request merged**